src/main/org/codehaus/groovy/runtime/DefaultGroovyMethods.java

```diff
@@ -5176,7 +5176,9 @@ public static Map groupBy(Object[] self, List<Closure> closures) {
             G key = outer.getKey();
             List<Map.Entry<K, V>> entries = outer.getValue();
             Map<K, V> target = createSimilarMap(self);
-            putAll(target, entries);
```

Comment:
my suggestion would be to change putAll to `public static <K, V> Map<K, V> putAll(Map<K, V> self, Collection<? extends Map.Entry<? extends K, ? extends V>> entries) `
which more or less adds `? extends` before the Map.Entry part and to change back to use the putAll instead of the loop